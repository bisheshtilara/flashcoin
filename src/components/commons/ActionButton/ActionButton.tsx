import { EyeIcon, TrashIcon } from "@heroicons/react/24/solid";
import React from "react";

const ActionButton: React.FC<{
  remove?: boolean;
  details?: boolean;
  action?: () => void;
}> = ({ remove = false, details = false, action = () => {} }) => {
  return (
    <div
      role="button"
      className={`py-1 px-2 rounded-lg ${
        details ? "bg-blue-500" : remove ? "bg-red-500" : null
      } flex items-center gap-2 text-white transition-all ease-in hover:translate-y-1`}
      onClick={action}
    >
      {details ? (
        <EyeIcon className="h-5" />
      ) : remove ? (
        <TrashIcon className="h-5" />
      ) : null}
      <p>{details ? "Details" : remove ? "Remove" : null}</p>
    </div>
  );
};

export default ActionButton;
