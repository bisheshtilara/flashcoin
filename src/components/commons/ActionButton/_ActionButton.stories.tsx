import { ComponentMeta, ComponentStory } from "@storybook/react";
import ActionButton from "./ActionButton";

export default {
  component: ActionButton,
} as ComponentMeta<typeof ActionButton>;


const defaultArgs: {
  remove?: boolean;
  details?: boolean;
  action?: () => void;
} = {
  remove: true,
  details: false,
  action: () => {},
};

const Template: ComponentStory<typeof ActionButton> = (args: {
  remove?: boolean;
  details?: boolean;
  action?: () => void;
}) => (
  <ActionButton {...args} />
);

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs,
};
