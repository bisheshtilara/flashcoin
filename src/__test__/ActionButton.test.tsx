import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ActionButton from "../components/commons/ActionButton";

describe("ActionButton", () => {
  it("should render correctly", () => {
    const { container } = render(<ActionButton />);
    expect(container).toMatchSnapshot();
  });

  it("should have text remove", () => {
    const { container } = render(<ActionButton remove />);
    expect(container.textContent).toBe("Remove");
  });

  it("should have text details", () => {
    const { container } = render(<ActionButton details />);
    expect(container.textContent).toBe("Details");
  });

  it("should appern alert when click on button", () => {
    const mockFn = jest.fn();
    const { container } = render(
      <ActionButton remove action={mockFn} />
    );
    fireEvent.click(screen.getByText("Remove"));
    expect(mockFn).toBeCalled();
  });
});
