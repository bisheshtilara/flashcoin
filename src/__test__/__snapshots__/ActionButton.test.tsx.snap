// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should render correctly 1`] = `
<div>
  <div
    class="py-1 px-2 rounded-lg null flex items-center gap-2 text-white transition-all ease-in hover:translate-y-1"
    role="button"
  >
    <p />
  </div>
</div>
`;
